### ISSUE TITLE
Billing Agreements not returning Plan Name / Description / Type - https://api.sandbox.paypal.com/v1/payments/billing-agreements/

### ISSUE THREAD

**Issue by puremana** [paypal/PayPal-REST-API-issues#292]
### General information

* Environment: Sandbox
* `PayPal-Debug-ID` values:
* Language, language version, and OS: * PHP 7.1.16 & PHP SDK version 1.14.0

### Issue description

The https://developer.paypal.com/docs/api/payments.billing-agreements/v1/#billing-agreements_get documentation clearly states that the Agreement object will return the Plan Name, Description and Type in the response.

![](https://i.gyazo.com/9f83c8b864bb0dffe824475049bd2a19.png)

This is not the case.

I receive everything from the sample response excluding plan name, description and type.

Please update the response to be accurate or return those variables (adding in Plan ID would also be very helpful).

**Comment by mipapo** [comment]
Would be great to get the Plan id to know what kind of plan the user subscribed to. In the moment I don't have an idea how to identify which plan was used to create the agreement.

**Comment by puremana** [comment]
@mipapo What I ended up doing was setting the agreement description to contain unique identifiers of the PayPal plan as we store PayPal plan ids and other details in our database.

**Comment by mipapo** [comment]
@puremana Good idea - already thought of it, but of course it would be nicer if Paypal would just add the Plan id to the agreement id. But well we all know Paypal and their bad API

**Comment by puremana** [comment]
Yes I was quite surprised when I noticed they don't return Plan ID as it is absolutely crucial to know once a user agrees to a billing agreement. I did find a few other issues relating to that though, which is why this issue is specifically geared towards PayPal fixing their API to return Plan name / id / type.

Working with this API has been a horrible experience and I highly doubt I will ever use it again

**Comment by mipapo** [comment]
@puremana  Just another idea: If you have an own identifier you could add it to the return_url during the creation of the billing plan

**Comment by puremana** [comment]
@mipapo I think doing it that way you may run into a problem where the user could create a fake return URL with their token in order to get a more valuable subscription. Unless you had identifiers that weren't prone to this of course (such as encrypted or random values), but in our case this wasn't possible
